src/vm/jitinterface.cpp

```diff
@@ -13666,6 +13675,17 @@ BOOL LoadDynamicInfoEntry(Module *currentModule,
 
             _ASSERTE(pMethod->IsNDirect());
             NDirectMethodDesc *pMD = (NDirectMethodDesc*)pMethod;
+
+            // Check if the NDirectTarget is marked for GC suppression at which point
+            // we must resolve the target at JIT time instead of at run time since there
+            // will be no GC frame and not in the correct mode.
+            void *tgtMaybe = nullptr;
+            if (NDirectMethodDesc::TryImportNDirectTargetForNoGCTransition(pMD, &tgtMaybe))
```

Comment:
This is performance sensitive path for startup with AOT images. `TryImportNDirectTargetForNoGCTransition` does a custom attribute lookups that is not a cheap operation. It would be really to figure out how to do this without the custom attribute lookups.

Reply:
Done.